src/net.cpp

```diff
@@ -887,7 +902,7 @@ bool CConnman::AttemptToEvictConnection()
                                                node->nLastBlockTime, node->nLastTXTime,
                                                HasAllDesirableServiceFlags(node->nServices),
                                                peer_relay_txes, peer_filter_not_null, node->addr, node->nKeyedNetGroup,
-                                               node->m_prefer_evict};
+                                               node->m_prefer_evict, node->addr.IsLocal()};
```

Comment:
I discovered that doing `IsLocal(node->addr)` is different from `node->addr.IsLocal()`, and the latter appears to be correct, but if someone can confirm that this captures the right idea I'd appreciate it.

Reply:
Somewhat confusingly:

- `node->addr.IsLocal()` returns true only if `node->addr` is IPv[46] and is in the loopback address ranges
- `IsLocal(node->addr)` checks whether or not `node->addr` resides in `mapLocalHost`, which tracks our view of addresses we might be referred to externally

So in this case, I believe `node->addr.IsLocal()` is correct.

Reply:
Yes. addr.IsLocal() is right. IsLocal() ought to be renamed to  IsProbablyMyAddress. :)